addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties

```diff
@@ -55,7 +55,7 @@ ascanbeta.elinjection.soln = Perform data validation best practice against untru
 ascanbeta.entityExpansion.desc = An exponential entity expansion, or "billion laughs" attack is a type of denial-of-service (DoS) attack. It is aimed at parsers of markup languages like XML or YAML that allow macro expansions.
 ascanbeta.entityExpansion.name = Exponential Entity Expansion (Billion Laughs Attack)
 ascanbeta.entityExpansion.other = The attack HTTP request received a response after {0} seconds.
-ascanbeta.entityExpansion.refs = https://en.wikipedia.org/wiki/Billion_laughs_attack\nhttp://projects.webappsec.org/XML-Entity-Expansion\nhttp://cwe.mitre.org/data/definitions/776.html
+ascanbeta.entityExpansion.refs = https://en.wikipedia.org/wiki/Billion_laughs_attack\nhttps://cse.google.com/cse?cx=partner-pub-9396229490951644%3A4ygj091ckzs&ie=ISO-8859-1&q=XML+Entity+Expansion&sa=Search&siteurl=www.webappsec.org%2F&ref=www.google.com%2F&ss=125j15625j2\nhttps://cse.google.com/cse?cx=partner-pub-9396229490951644%3A4ygj091ckzs&ie=ISO-8859-1&q=XML+Entity+Expansion&sa=Search&siteurl=www.webappsec.org%2F&ref=www.google.com%2F&ss=125j15625j2\nhttps://cwe.mitre.org/data/definitions/776.html
```

Comment:
You can just drop the middle link if it isn't available via https. I'm not confident that google cache links are stable.

You could also use: https://owasp.org/www-community/vulnerabilities/XML_External_Entity_(XXE)_Processing

Reply:
Okay, now to the actual work at hand 🙂